src/App.tsx

```diff
@@ -1,16 +1,45 @@
-import React from "react";
+import React, {useEffect, useState} from "react";
 
 import Header from "./containers/Header/Header";
 import Main from "./containers/Main/Main";
 import Footer from "./containers/Footer/Footer";
+import ModalWindow from "./components/ModalWindow/ModalWindow";
 import "./App.css";
+import {initialState} from "./assets/data/constData";
+import {Movie, State} from "./models/movie";
+import {fetchMovies} from "./api/movieService";
 
 const App: React.FC = () => {
+  const [openModal, setOpenModal] = useState(false);
+  const [mode, setMode] = useState('');
+  const [initialDate, setInitialDate] = useState<State | Movie>(initialState);
+  const [movies, setMovies] = useState<Movie[] | []>([]);
```

Comment:
```suggestion
  const [movies, setMovies] = useState<Movie[]>([]);
```
Array of `Movie` is still an array.